ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/recovery/RecoverySocketUnitTest.java

```diff
@@ -0,0 +1,178 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package com.hp.mwtests.ts.arjuna.recovery;
+
+import com.arjuna.ats.arjuna.common.recoveryPropertyManager;
+import com.arjuna.ats.arjuna.recovery.RecoveryDriver;
+import com.arjuna.ats.arjuna.recovery.RecoveryManager;
+import org.jboss.logging.Logger;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * Test cases which work with a direct connection to socket where RecoveryManager listens at.
+ */
+public class RecoverySocketUnitTest {
+    private static final Logger log = Logger.getLogger(RecoverySocketUnitTest.class);
+
+    private static boolean socketRecoveryListenerInitialState;
+    private static int periodicRecoveryPeriodInitialState, recoveryBackoffPeriodInitialState;
+
+    private InetAddress recoveryManagerHost = null;
+    private int recoveryManagerPort = 0;
+    private RecoveryManager recoveryManager;
+
+    @BeforeClass
+    public static void getInitialState() {
+        socketRecoveryListenerInitialState = recoveryPropertyManager.getRecoveryEnvironmentBean().isRecoveryListener();
+        periodicRecoveryPeriodInitialState =  recoveryPropertyManager.getRecoveryEnvironmentBean().getPeriodicRecoveryPeriod();
+        recoveryBackoffPeriodInitialState = recoveryPropertyManager.getRecoveryEnvironmentBean().getRecoveryBackoffPeriod();
+    }
+
+    @AfterClass
+    public static void returnBackInitialState() {
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryListener(socketRecoveryListenerInitialState);
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setPeriodicRecoveryPeriod(periodicRecoveryPeriodInitialState);
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryBackoffPeriod(recoveryBackoffPeriodInitialState);
+    }
+
+    @Before
+    public void enableRecoveryListener() throws InterruptedException{
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryListener(true);
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setPeriodicRecoveryPeriod(1);
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryBackoffPeriod(1);
+        recoveryManager = RecoveryManager.manager();
+        // wait for the socket would be established
+        Thread.sleep(50);
```

Comment:
Will you add more comment about why the test is waiting?

Reply:
Thanks. Good point. This is my omission from the prior test setup.
The recovery manager startup is synchronous and the waiting should not be necessary here at all.